ner_processing/__main__.py

```diff
@@ -38,39 +39,91 @@ def getLineCount(filepaths: List[str]) -> int:
+
+    if seconds < 0:
+        minutes -= 1
+        seconds += 60
+
+    print("Time to process (Minutes:Seconds.Microseconds): " + str(minutes) + ":" + str(seconds) + "." + str(
+        microseconds))
+
+
 if __name__ == "__main__":
+    """
+    Processes the log files in the log folder and puts them in the output.csv file
+    """
+
+    start_time = datetime.now().strftime("%M:%S:%f").split(":")
     line_count = getLineCount([LOGS_DIRECTORY + name for name in listdir(LOGS_DIRECTORY)])
     current_line = 0
+    manager = multiprocessing.Manager()
+    return_dict = manager.dict()
     print(f"Processing a total of {line_count} lines")
-
-    processed_data: List[Data] = []
-    for fp in listdir(LOGS_DIRECTORY):
-        with open(LOGS_DIRECTORY + fp) as file:
-            line_num = 0
-            for line in file:
-                line_num += 1
-                current_line += 1
-                if current_line % 5000 == 0:
-                    print(f"Line {line_num}")
-                try:
-                    message: Message = processLine(line, FORMAT)
-                    processed_data.extend(message.decode())
-                except:
-                    print(f"Error with line {line_num} in file {file.name}")
-                    pass
+    processors = 8
+    process_chunk_size = 85000  # Processes data in chunks, specified by this variable
 
     print(f"Writing to the CSV")
     header = ["time", "data_id", "description", "value"]
+
     with open(OUTPUT_PATH, "w", encoding="UTF8", newline="") as f:
         writer = csv.writer(f)
         writer.writerow(header)
-        for data in processed_data:
-            str_time = data.timestamp.toString("yyyy-MM-ddTHH:mm:ss.zzzZ")
-            writer.writerow([str_time, data.id, DATA_IDS[data.id], data.value])
 
+        for fp in listdir(LOGS_DIRECTORY):
+            with open(LOGS_DIRECTORY + fp) as file:
+                line_num = 0
+                lines = []
+                for line in file:
+                    line_num += 1
+                    current_line += 1
+                    if current_line % 5000 == 0:
+                        print(f"Line {line_num}")
 
+                    lines.append(line)
+                    try:
+                        if line_num % process_chunk_size == 0:
+                            """
+                            When stored data reaches specified amount, use threads to decode data faster
+                            """
 
+                            with multiprocessing.Pool(processors) as p:
+                                out = p.map(thread, lines)
+                                lines = []
+                            for data in out:
+                                for sub_data in data:
+                                    str_time = sub_data.timestamp.toString("yyyy-MM-ddTHH:mm:ss.zzzZ")
+                                    writer.writerow([str_time, sub_data.id, DATA_IDS[sub_data.id], sub_data.value])
```

Comment:
Throw these lines in a seperate function (something that looks like `process_lines(lines, writer)`) to remove the duplication.

Reply:
This was a great suggestion, it forced me to use lines.clear(), which sped up the program by 3-4% with my test file.